contracts/strategies/stkCvxFXS/Strategy.sol

```diff
@@ -0,0 +1,133 @@
+        IERC20(CVXFXS_TOKEN).safeApprove(address(cvxFxsStaking), 0);
+        IERC20(CVXFXS_TOKEN).safeApprove(
+            address(cvxFxsStaking),
+            type(uint256).max
+        );
+    }
+
+    /// @notice Update the harvester contract
+    /// @param _harvester address of the new contract
+    function setHarvester(address _harvester) external onlyOwner {
+        require(_harvester != address(0));
+        harvester = _harvester;
+        // ensures all rewards are redirected to harvester
+        // if regular claim reward is triggered on staking contract
+        cvxFxsStaking.setRewardRedirect(harvester);
+    }
+
+    /// @notice Query the amount currently staked
+    /// @return total - the total amount of tokens staked
+    function totalUnderlying() public view returns (uint256 total) {
+        return cvxFxsStaking.balanceOf(address(this));
+    }
+
+    /// @notice Deposits all underlying tokens in the staking contract
+    function stake(uint256 _amount) external onlyVault {
+        cvxFxsStaking.stake(_amount);
+    }
+
+    /// @notice Withdraw a certain amount from the staking contract
+    /// @param _amount - the amount to withdraw
+    /// @dev Can only be called by the vault
+    function withdraw(uint256 _amount) external onlyVault {
+        cvxFxsStaking.withdraw(_amount);
+        IERC20(CVXFXS_TOKEN).safeTransfer(vault, _amount);
+    }
+
+    /// @notice Claim rewards and swaps them to cvxFXS for restaking
+    /// @dev Can be called by the vault only
+    /// @param _caller - the address calling the harvest on the vault
+    /// @param _minAmountOut - min amount of cvxFxs expected
+    /// @return harvested - the amount harvested
+    function harvest(address _caller, uint256 _minAmountOut)
+        external
+        onlyVault
+        returns (uint256 harvested)
+    {
+        // claim rewards
+        cvxFxsStaking.getReward(address(this), harvester);
+
+        uint256 _cvxFxsBalance = IHarvester(harvester).processRewards();
+        require(_cvxFxsBalance >= _minAmountOut, "slippage");
+
+        uint256 _stakingAmount = _cvxFxsBalance;
+
+        if (_cvxFxsBalance > 0) {
+            // if this is the last call, no fees
+            if (IGenericVault(vault).totalSupply() != 0) {
+                // Deduce and pay out incentive to caller (not needed for final exit)
+                if (IGenericVault(vault).callIncentive() > 0) {
+                    uint256 incentiveAmount = (_cvxFxsBalance *
+                        IGenericVault(vault).callIncentive()) / FEE_DENOMINATOR;
+                    IERC20(CVXFXS_TOKEN).safeTransfer(_caller, incentiveAmount);
+                    _stakingAmount = _stakingAmount - incentiveAmount;
+                }
+                // Deduce and pay platform fee
+                if (IGenericVault(vault).platformFee() > 0) {
+                    uint256 feeAmount = (_cvxFxsBalance *
+                        IGenericVault(vault).platformFee()) / FEE_DENOMINATOR;
```

Comment:
If you're not planning on having many different cvxFXS vaults associated with this single strategy, it would be more gas efficient to have values like this stored within this contract (vs. doing an external call).